holidays/countries/united_states.py

```diff
@@ -110,479 +109,869 @@ class UnitedStates(HolidayBase):
+
+        # Good Friday
+        self._add_good_friday("Good Friday")
+
+        # San Jacinto Day
+        if self._year >= 1875:
+            self._add_holiday("San Jacinto Day", APR, 21)
+
+        # Emancipation Day In Texas
+        if self._year >= 1980:
+            self._add_holiday("Emancipation Day In Texas", JUN, 19)
+
+        # Lyndon Baines Johnson Day
+        if self._year >= 1973:
+            self._add_holiday("Lyndon Baines Johnson Day", AUG, 27)
+
+        # Friday After Thanksgiving
+        if self._year >= 1975:
+            self._add_holiday(
+                "Friday After Thanksgiving",
+                _get_nth_weekday_of_month(4, THU, NOV, self._year) + td(days=+1),
+            )
+
+        # Christmas Eve
+        if self._year >= 1981:
+            self._add_christmas_eve_holiday()
 
         # Day After Christmas
-        dt = date(year, DEC, 26)
-        if self.subdiv == "NC" and year >= 2013:
-            name = "Day After Christmas"
-            self[dt] = name
-            # If on Saturday or Sunday, observed on Monday
-            if self.observed and self._is_weekend(dt):
-                self[_get_nth_weekday_from(1, MON, dt)] = f"{name} (Observed)"
-            # If on Monday, observed on Tuesday
-            elif self.observed and self._is_monday(dt):
-                self[dt + td(days=+1)] = f"{name} (Observed)"
-        elif self.subdiv == "TX" and year >= 1981:
-            self[dt] = "Day After Christmas"
-        elif self.subdiv == "VI":
-            self[dt] = "Christmas Second Day"
+        if self._year >= 1981:
+            self._add_christmas_day_two("Day After Christmas")
 
-        # New Year's Eve
-        if (self.subdiv in {"KY", "MI"} and year >= 2013) or (
-            self.subdiv == "WI" and year >= 2012
-        ):
-            self._add_with_observed(
-                date(year, DEC, 31), "New Year's Eve", before=True, after=False
+    def _add_subdiv_um_holidays(self):
+        pass
+
+    def _add_subdiv_ut_holidays(self):
+        # Pioneer Day
+        if self._year >= 1849:
+            self._add_observed(self._add_holiday("Pioneer Day", JUL, 24))
+
+    def _add_subdiv_va_holidays(self):
+        # Lee Jackson Day
+        if 1889 <= self._year <= 2020:
+            if self._year >= 2000:
+                dt = _get_nth_weekday_of_month(3, MON, JAN, self._year) + td(days=-3)
+            elif self._year >= 1983:
+                dt = _get_nth_weekday_of_month(3, MON, JAN, self._year)
+            else:
+                dt = date(self._year, JAN, 19)
+            self._add_holiday("Lee Jackson Day", dt)
+
+        # Inauguration Day
+        if self._year >= 1789 and (self._year - 1789) % 4 == 0:
+            dt = (JAN, 20) if self._year >= 1937 else (MAR, 4)
+            self._add_observed(self._add_holiday("Inauguration Day", *dt), before=False)
+
+    def _add_subdiv_vi_holidays(self):
+        # Three King's Day
+        self._add_epiphany_day("Three King's Day")
```

Comment:
```suggestion
        self._add_epiphany_day("Three Kings Day")
```
Three Kings' Day would also work.